flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/TestJobExecutor.java

```diff
@@ -164,22 +164,23 @@ private void waitForFailover(BlockingQueue<TestEvent> queue) throws Exception {
     }
 
     private void handleFailoverTimeout(TimeoutException e) throws Exception {
+        JobStatus jobStatus = miniClusterResource.getClusterClient().getJobStatus(jobID).get();
         String message =
                 String.format(
                         "Unable to failover the job: %s; job status: %s",
-                        e.getMessage(),
-                        miniClusterResource.getClusterClient().getJobStatus(jobID).get());
-        Optional<SerializedThrowable> throwable =
-                miniClusterResource
-                        .getClusterClient()
-                        .requestJobResult(jobID)
-                        .get()
-                        .getSerializedThrowable();
-        if (throwable.isPresent()) {
-            throw new RuntimeException(message, throwable.get());
-        } else {
-            throw new RuntimeException(message);
+                        e.getMessage(), jobStatus);
+        if (jobStatus.isGloballyTerminalState()) {
+            Optional<SerializedThrowable> throwable =
```

Comment:
```suggestion
            // Only searching for the throwable in case of a terminal state prevents timeouts when waiting for the job result.
            Optional<SerializedThrowable> maybeThrowable =
```

Reply:
Would it make sense to check for `jobStatus == JobStatus.FAILED` instead?

Reply:
I think it's better to keep `isGloballyTerminalState` condition because it might then report `JobCancellationException` if `jobStatus == CANCELED` or something else for `UNKNOWN`, which might be useful.
(and as long as job is finished and the API returns `Optional`, no exception shouldn't be a problem)